---
output:
  pdf_document:
    latex_engine: xelatex
header-includes:
    - \usepackage{fancyhdr}
    - \usepackage{fontspec}
    - \usepackage{xcolor}
mainfont: Roboto
sansfont: Roboto
urlcolor: badgerred
---
\definecolor{badgerred}{RGB}{197,5,12}
\addtolength{\headheight}{3.0cm}
\pagestyle{fancy}
\lhead{\raisebox{-0.25\height}{\includegraphics[height=2.5cm]{/assets/wslh_header.png}}}
\rhead{
\Huge Sequencing Analysis Report\\
\Large `r paste(Sys.Date())`
}
\renewcommand{\headrulewidth}{1pt}
\renewcommand{\headrule}{\hbox to\headwidth{%
    \color{badgerred}\leaders\hrule height \headrulewidth\hfill}}

```{r include=FALSE}
#testing
## Libraries
library(superheat)
library(ggtree)
library(phangorn)

date <- Sys.Date()

## Figure size
knitr::opts_chunk$set(fig.width=9, fig.height=7)

```

### *`r species`* Analysis

The analysis of the *`r species`* samples you provided us (n = `r num.iso`) has been completed and this is
what we were able to determine:

### SNP Heatmap

The number of Single Nucleotide Polymorphisms (SNPs) between each sample is shown on the heatmap below. There is no hard and fast rule for determining how many SNPs are needed to classify an outbreak. Generally it is best to look for patterns in the data between both the SNP data and the core-genome tree.

```{r echo=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.align="left")
set.seed(123)
colnames(snp.mat) <- row.names(snp.mat)

mat_data <- as.matrix(snp.mat)

# set the text colors
snp.col <- scale(mat_data) < .0000001
# set all values that satisfy the condition to "white"
snp.col <- gsub("TRUE", "white", snp.col)
# set all values that do not satisfy the condition to "black"
snp.col <- gsub("FALSE", "black", snp.col)
# convert to matrix
snp.col <- matrix(snp.col, ncol = ncol(mat_data))

superheat(mat_data,
          pretty.order.rows = TRUE,
          pretty.order.cols = TRUE,
          row.dendrogram = TRUE,
          left.label.col = "white",
          bottom.label.col = "white",
          bottom.label.text.angle = 90,
        #  heat.pal=pal,
          X.text = round(mat_data, 1),
          X.text.size = 4,
          left.label.text.size = 4,
          bottom.label.text.size = 4,
          left.label.size = .75,
          bottom.label.size = .75,
          X.text.col = snp.col,
         # extreme.values.na = FALSE,
          legend.num.ticks = 4,
          grid.hline = FALSE,
          grid.vline = FALSE,
          padding = 0,
          legend.vspace = 0.05)
```
\newpage

### Core-genome phylogenetic tree

The core-genome is the core set of genes shared across all isolates in the sample. Using variation within that core set of genes, we can estimate how related isolates are. We do this by determining if isolates share a similar common ancestor. Here we are looking for isolates that cluster together and share a small amount of horizontal distance on the tree. The numbers in boxes are the bootstrap values, a value greater than 95 indicates a robustly supported grouping of isolates.

```{r echo=FALSE, message=FALSE, warning=FALSE}
tree <-  read.tree(nwk)
mpt <- midpoint(tree)
ggtree(mpt) + geom_label2(aes(label=label, x = branch, subset = !isTip)) + geom_tiplab() + xlim(-.0001, .006) + geom_treescale(offset = .1)
```

### Methods

The figures shown here were generated using sequence data processed with our data analysis pipeline, Dryad ([https://github.com/k-florek/dryad](https://github.com/k-florek/dryad)). SNPs were called using the Center for Food Safety and Applied Nutrition ([CFSAN](https://github.com/CFSAN-Biostatistics/snp-pipeline)) SNP pipeline within Dryad. The core-genome tree was generated using an alignment of coding regions shared across 99% of isolates. The tree was generated using a maximum-likelihood method with the GTR+G substitution model and bootstrapped with 1000 replicates. Report generated using R and designed by Abigail Shockey.
